blog/flexdashboard-youtube-api/index.md

```diff
@@ -0,0 +1,356 @@
+
+This code builds out the sidebar section. I created a list of each R-Ladies chapter that I was able to find and arranged them by name. With `shiny::HTML()`, the dashboard can render the URLs as HTML (the reason for all that manipulation earlier on).
+
+```` R
+Sidebar {.sidebar}
+-----------------------------------------------------------------------
+
+The purpose of this dashboard is to provide a running feed of R-Ladies videos posted to YouTube. It is refreshed every six hours.
+
+Currently, the feed includes these channels:
+
+```{r}
+dat_join %>% 
+  arrange(chapter) %>% 
+  distinct(channel_image_url) %>% 
+  pull() %>% 
+  shiny::HTML()
+```
+````
+
+For the main body, I used the [DT](https://rstudio.github.io/DT/) package to create a table for the information in our clean dataset. I added a bit of HTML to make it expand to the entire column height.
+
+A couple of fun tricks to notice:
+
+* `escape = FALSE` renders the URLs within the table as HTML
+* The list within `options` aligns the text within columns (in this case, to be in the middle of the cell for all columns)^[I found out about this here: [https://stackoverflow.com/questions/35749389/column-alignment-in-dt-datatable](https://stackoverflow.com/questions/35749389/column-alignment-in-dt-datatable).]
+
+There are a lot of customization options!
+
+```` R
+Column {data-width=900}
+-----------------------------------------------------------------------
+
+### By default, the list is sorted by latest video.
+
+<style>
+.dataTables_scrollBody {
+    max-height: 100% !important;
+}
+</style>
+
+```{r}
+dat_dashboard_dat %>%
+  select(-chapter, -channel_image_url) %>%
+  datatable(
+    colnames = c("Date", "Channel", "Video"),
+    filter = "top",
+    escape = FALSE,
+    height = "1000",
+    options = list(columnDefs = list(
+      list(className = "dt-middle", targets = "_all")
+    ))
+  )
+```
+````
+
+And that's it! See the [final dashboard code](https://github.com/ivelasq/rladies-video-feed/blob/main/index.Rmd) on GitHub.
+
+![Final dashboard](dashboard.png){fig-alt="Screenshot of the final dashboard with a sidebar of the R-Ladies channels and the main section showing the table of videos from the various chapters"}
+
+[Try it out](https://ivelasq.github.io/rladies-video-feed/) --- search "Shiny" to see any video with Shiny in the title, or "Ecuador" to see all the videos from R-Ladies Ecuador!
+
+## Schedule updates
+
+I am glad to have a dashboard with a list of R-Ladies videos. But, what about future ones?
+
+One option to keep your dashboard updated is to use [GitHub Actions](https://github.com/features/actions). GitHub Actions are saved as a `.yml` file within a folder called `.github/workflows/`. It specifies the steps to take based on a certain action (with every push, every 12 hours, etc.). By setting up a 'workflow', you can automate your workflow and refresh the data on a schedule.
+
+h/t to [@gvelasq](https://github.com/gvelasq) for this GitHub Action:
```

Comment:
Suggest change to:

h/t to [@gvelasq](https://github.com/gvelasq) and this amazing [example](https://github.com/r-lib/actions/blob/v2/examples/render-rmarkdown.yaml) in the [r-lib/actions](https://github.com/r-lib/actions/tree/v2/examples#readme) repo for this GitHub Action: